packages/hash/frontend/src/components/BlockLoader/BlockLoader.tsx

```diff
@@ -56,11 +56,10 @@ export const BlockLoader: VoidFunctionComponent<BlockLoaderProps> = ({
   linkGroups,
   linkedEntities,
   linkedAggregations,
+  routeHash,
   shouldSandbox,
   sourceUrl,
 }) => {
-  const router = useRouter();
```

Comment:
Hey @nathggns, for context, the Asana comment on which this PR is based:

> BlockLoader was that its a weird component to depend on the router. You could have it take a prop called "scrollIntoView", and the logic of whether it should be scrolled into view can be checked outside of the component

Right now, I've tried to move the router out of this component to at least remove that dependency from this component.

I understand from your comment that the actual goal is to have 0 router dependency inside of `BlockLoader`, including not having to know what the current route's hash is.

However, the current scroll function depends on whether the block has loaded in or not to determine when it should scroll to the block. This makes it difficult to make the component not depend on the route hash.

We could come back to extracting the complete scroll function out of the component later, when it's more relevant.

Reply:
> However, the current scroll function depends on whether the block has loaded in or not to determine when it should scroll to the block

My understanding is that this isn't true. It waits until the block has loaded to trigger the scroll, which is why the effect is here in the first place, but it ought to know before the block is loaded whether it will scroll. This is determined by checking the route hash against the entityId, both of which are available before the block is loaded. 

To be honest, I'm not sure that this PR is worth merging if this is the extent of the update. I appreciate the work you've done, but the idea is to move the "is this the active block" code to the top level, but if all we've done is move where the route hash is obtained without actually moving the logic, all this does is make the code less clear without the benefit of the component not depending on the URL 

I think the ideal solution is `BlockLoader` passing the block load event up to a central `BlockLoadProvider` context which holds a handler meaning `PageBlock` can handle the actual scrolling / have the dependency on the router. Let me know if you want to pair on this.

Reply:
Our call was pretty helpful. All of the scrolling code has been removed from here and moved into the `BlockLoadedContext` instead.